homeassistant/components/p1_monitor/const.py

```diff
@@ -0,0 +1,229 @@
+"""Constants for the P1 Monitor integration."""
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    CURRENCY_EURO,
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.util import dt
+
+DOMAIN: Final = "p1_monitor"
+LOGGER = logging.getLogger(__package__)
+DEFAULT_TIME_BETWEEN_UPDATE = 5
+
+ATTR_ENTRY_TYPE: Final = "entry_type"
+ENTRY_TYPE_SERVICE: Final = "service"
+
+CONF_TIME_BETWEEN_UPDATE = "time_between_update"
+
+SERVICE_SMARTMETER: Final = "smartmeter"
+SERVICE_PHASES: Final = "phases"
+SERVICE_SETTINGS: Final = "settings"
+
+SERVICES: dict[str, str] = {
+    SERVICE_SMARTMETER: "SmartMeter",
+    SERVICE_PHASES: "Phases",
+    SERVICE_SETTINGS: "Settings",
+}
+
+SENSORS: dict[str, list[SensorEntityDescription]] = {
```

Comment:
The common practice is to use a tuple instead of list for these blocks of data. They take somewhat less space since they are immutable.

Reply:
Tried to adjust this, but then the values are no longer visible in Home Assistant, I see that [Ambee](https://github.com/home-assistant/core/blob/dev/homeassistant/components/ambee/const.py#L37) use the same approach on this.

Reply:
It only gives me error messages from mypy when I change this 🤔 

![image](https://user-images.githubusercontent.com/20448157/129735245-7d1714b1-382f-42d0-b84b-a320f77b9b82.png)

Reply:
tuple[SensorEntityDescription, ...]